### ISSUE TITLE
Programmatic Interface for "Save" menu option

### ISSUE THREAD

**Issue by bieniekmateusz** [rgerum/pylustrator#10]
Thanks for creating the package, it's really nice. 

I was thinking of adding your package to another software (PyMOL) and it would be really useful to be able to define pylustrator's "Save" option. I don't work with IPythonNotebook. 

Thanks, Mat

**Comment by rgerum** [comment]
Hi Mat,
I am glad that you like the pylustrator package. But I am not 100% sure what your request is about. You want a custom callback that can be defined, you your script gets notified when pylustrator saves the code? For what purpose? To alter the code that pylustrator generated? Or to store it in a different file?
I googled your PyMOl and it seems that it is a user interface to display molecules. I am not quite sure at which point you want to include Pylustrator.

**Comment by bieniekmateusz** [comment]
Hi @rgerum 

Thanks for the reply. I was thinking about using it to save it to a different file. It is basically a plot manager for PyMOL. In a similar way I creates the plot and allows the user to further adjust the script later. 

Basically, together with a colleague we have been working on an extension that allows to carry out additional analysis which in turn can be plotted using matplotlib. We have implemented very simple interactive features, but it is nothing in comparison to what you have done here. With a simple API it would be easy to plug-in your package. 

Cheers, Mat

**Comment by rgerum** [comment]
Hi Mat,
so I am not 100% sure what your suggestion should contain.
You want something like `pylustrator.start(target_filename)` ? as a command? Would this be enough to implement the feature you like?
Do you just want to style the plot in the interactive way or do you also want the use to plot different parts of the dataset? Currently pylustrator "just" supports the styling. So I am not really sure what you would gain if the changes pylustrator gathered form the user interaction are stored in a new separate file.

**Comment by bieniekmateusz** [comment]
Hi @rgerum,

I tried again using your software and I think I misused it slightly. I opened it inside of an ipython interpreter and for that reason that save button did not do anything for me. However, now I just tried opening a script with your pylustrator and it updates properly the file. In PyMOL I actually open the file so that means that your code would work fine. 

So sorry for the misunderstanding, I think the issue can be closed. 

Thanks, Mat
